src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    const clearedArray = this
+      .map((value) => value === undefined || value === null ? '' : value);
+
+    if (clearedArray.length === 0) {
```

Comment:
не потрібно перевіряти на `0`
```suggestion
    if (clearedArray.length) {
```